### ISSUE TITLE
Feature Request: Custom Contracts via BTDEX

### ISSUE THREAD

**Issue by Valareos** [btdex/btdex#222]
I am increasingly getting requests to convert crypto not supported on BTDEX into Signa. However. it requires a trust relationship to do.

Trustless is always the way to go, so I believe teh BTDDEX program, with minor changes, can support custom contracts. In fact, it may be able to replace some of the less used coin pairs

Step 1: Person sending the signa must start the contract. The signa has to be locked by Smart Contract to make it work.  But, they can fill out all the information required, such as how much crypto and what address. In addition there is a new "private pin" section to ensure only the intended recipient can take the contract

Step 2: recipient, armed with the Contract ID and the private pin, accepts the contract.  Now, by the same rules as other coin pairs, they got 2 hours to send the crypto.

Step 3: The one sending the signa has 24 hours to confirm the contract, as is the current standard.

In other words, it works the same as any other coin pair or contract, and uses all the same programming with 2 exceptions

1: custom contract fields
2: Private Pin to ensure intended recipient is used

theoretically, 2 is optional, because the recipient still has to pay the security deposit.

This simple change will make it easier for non standard pair swaps to happen
